pkg/util/kubernetes/autoscalers/datadogexternal_util.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoscalers
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	le "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection/metrics"
+)
+
+type minRemainingRequests struct {
```

Comment:
The class is pretty generic and should be called something like `minTracker`. You should also take in the target gauge as parameter instead of targeting `rateLimitsRemainingMin` directly.

Reply:
I agree, `minTracker` would make more sense as a name here.

However, I will remove the gauge `rateLimitsRemainingMin` completely from the struct, because passing it as a parameter will still create a dependency on the tags that need to be set with the gauge. I think it would be better to set the gauge externally (i.e outside the update method)